docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+
+  Ad magna labore tempor dolor dolore nulla magna Lorem
+  magna qui et. Ipsum eiusmod minim ad est nostrud sunt occaecat.
+  <a name="end:my-block"></a>
+  ```
+
+**Implicit eblocks** are defined by any anchor whose name is not starting with `begin:` or `end:`, and the point where the nesting level of the first non-immediate AST node either remains on the same level or decreases in comparison to the starting place of the anchor. The nesting level is changed e.g. with different levels of headings and bullet list indents. Additionally, the eblock can also be started with a heading, in which case the eblock ends when the chapter following the heading ends. In case the end of a file is found before the nesting level of the markdown content decreases, eblock ends there. The example below illustrate implicit eblocks with various eblock definitions.
+
+```markdown
+# My Note
+
+Occaecat esse minim Lorem fugiat mollit eu eiusmod magna dolor.
+Velit ea cupidatat pariatur amet pariatur est est laboris culpa
+ipsum magna eu nulla esse.
+
+## Second Level
+
+<a name="y"></a>
+Cupidatat velit esse est anim consequat enim ad.
+
+- Nostrud incididunt
+- <a name="x"></a> consequat non dolore
+  - consequat fugiat
+  - eiusmod esse ipsum
+    - anim nulla aute
+- ad reprehenderit
+
+## Another Second Level
+
+Ut reprehenderit sint quis pariatur. Non ullamco eiusmod nulla
+do sunt pariatur magna fugiat. Ullamco nulla magna nostrud
+in aute eu laborum ex est nostrud nostrud excepteur Lorem.
+```
+
+Examples of implicit eblocks defined above:
+
+- `my-note` refers to the whole note (id of the H1 heading of the note).
+- `second-level` refers to the "Second Level" chapter (stopping before the heading "Another Second Level")
+- `y` refers to the text `Cupidatat velit esse est anim consequat enim ad.`
+- `x` refers to the following list fragment:
+  ```markdown
+  - consequat non dolore
+    - consequat fugiat
+    - eiusmod esse ipsum
+      - anim nulla aute
+- `another-second-level` refers to the "Another Second Level" chapter (stopping at the end of the file)
+
+### Aliases
+
+Aliases define alternative names for notes. They are defined in an optional front matter, using an attribute "aliases". The example below shows how this can be done using YAML front matter:
+
+```YAML
+---
+aliases:
+  - My First Alias
+  - My Second Alias
+---
+```
+
+### Tags
+
+Tag is a sequence of non-whitespace characters prefixed with the hash character (e.g. `#my-hash`). Tags can work in one of two ways: 1) it's an abstract symbol that can be searched and inspected across notes (i.e. *abstract tag*) or 2) it's like 1, but simultaneously it's also a note in its own right (i.e. *file-based tag*). In the case of 1, it acts like hashtags in popular social media platforms like Instagram or Twitter. In the case of 2, it's more like a note where the user can write just like into any other note. These notes are stored in the `tags/`-folder within the Foam workspace.
```

Comment:
What if tags were also anchors? This sounds a bit crazy, but hear me out.

- The hashtag syntax is already mirrored in markdown link syntax `[](#tag)`
- The hashtag syntax is reminiscent of headings (`# Heading` vs `#tag`), which are already anchors by their own right
- In the compilation step, all we would need to do is to compile each tag to an anchor: `<a name="tag"></tag>`, which makes intuitive sense
  - Alternatively, they could be rendered into the document. (`<a name="tag">#tag</tag>`), and we could use use it as a "backlink" that shows everywhere it's being referenced, e.g. by showing a list of referrers as a popup on hover! 🤯
- They're somewhat pleasant to look at:

```
- some content here
  - this block needs linking to #block
    - children are part of...
    - the block scope
```

As to uniqueness, we could consider the semantics to be same as with multiple identical name-anchors: It points to the first one.  For embeds to tag-anchors, we could do a multi-embed where we embed all references to that tag within the document.

Again, this would require a build step to work in a published Foam, but maybe this is the "future better syntax" you referred to above?